modules/accounts/calendar.nix

```diff
@@ -51,20 +51,27 @@ let
         description = "The URL of the storage.";
       };
 
+      urlCommand = mkOption {
+        type = types.nullOr (types.listOf types.str);
+        default = null;
+        example = [ "~/get-url.sh" ];
+        description = "A command that prints the URL of the storage.";
+      };
+
       userName = mkOption {
         type = types.nullOr types.str;
         default = null;
         description = "User name for authentication.";
       };
 
-      # userNameCommand = mkOption {
-      #   type = types.nullOr (types.listOf types.str);
-      #   default = null;
-      #   example = [ "~/get-username.sh" ];
-      #   description = ''
-      #     A command that prints the user name to standard output.
-      #   '';
-      # };
+      userNameCommand = mkOption {
```

Comment:
I guess `userName` and `userNameCommand` are mutually exclusive. I.e., we should have an assertion that causes an error if the user has given both. Same holds for `url` and `urlCommand`.

Reply:
Added.